### PR TITLE
Dmp 1289 transcription implement authorisation for request body

### PR DESCRIPTION
**Pull request by karen-hedges**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-1289

### Change description ###

Added any context authentication

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/transcriptions/controller/TranscriptionController.java [hunk 1]
```diff
@@ -164,7 +167,7 @@ private boolean transcriptionTypesThatRequireDates(Integer transcriptionTypeId)
     }
 
     private boolean transcriptionDatesAreSet(OffsetDateTime startDateTime, OffsetDateTime endDateTime) {
-        return !isNull(startDateTime) || !isNull(endDateTime);
+        return !isNull(startDateTime) && !isNull(endDateTime);
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/transcriptions/controller/TranscriptionController.java:170]: I guess `nonNull` like in method validateTranscriptionRequestValues might be cleaner if needing to use Objects at all

File: src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/TranscriptionIdControllerAuthorisationImpl.java [hunk 2]
```diff
@@ -32,18 +32,14 @@ public ContextIdEnum getContextId() {
...
-
-        if (transcriptionIdParamOptional.isEmpty()) {
-            transcriptionIdParamOptional = Optional.ofNullable(request.getParameter(TRANSCRIPTION_ID_PARAM));
-            checkAuthorisationByTranscriptionId(transcriptionIdParamOptional, roles);
-        }
-
-        if (transcriptionIdParamOptional.isEmpty()) {
-            transcriptionIdParamOptional = Optional.ofNullable(request.getHeader(TRANSCRIPTION_ID_PARAM));
+        Optional<String> transcriptionIdParamOptional = getEntityParamOptional(request, TRANSCRIPTION_ID_PARAM);
+        if (transcriptionIdParamOptional.isPresent()) {
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/TranscriptionIdControllerAuthorisationImpl.java:42]: No need to have this isPresent line duplicated here if we have it also in the method checkAuthorisationByTranscriptionId line #63?

File: src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/TranscriptionIdControllerAuthorisationImpl.java [hunk 3]
```diff
@@ -32,18 +32,14 @@ public ContextIdEnum getContextId() {
         return TRANSCRIPTION_ID;
     }
 
+    public String getEntityIdParam() {
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/TranscriptionIdControllerAuthorisationImpl.java:35]: Are you missing `@Override` on this method?

Comment (karen-hedges) [reply src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/TranscriptionIdControllerAuthorisationImpl.java:35]: No, its not in the interface

**Review by richardbull — APPROVED** [review]

**Pull request merged**